File: frontend/lib/contracts/leaderboard.ts
/**
 * Leaderboard contracts.
 */

export interface LeaderboardEntry {
  rank: number
  user_id: number
  display_name: string
  total_xp: number
  current_streak: number
  is_current_user: boolean
}

export interface LeaderboardResponse {
  ranking_basis: string
  entries: LeaderboardEntry[]
  current_user: LeaderboardEntry
}

File: backend/app/schemas/leaderboard.py
"""Leaderboard response schemas."""
from pydantic import BaseModel, Field


class LeaderboardEntry(BaseModel):
    """Single ranked leaderboard row."""

    rank: int
    user_id: int
    display_name: str
    total_xp: int
    current_streak: int
    is_current_user: bool


class LeaderboardResponse(BaseModel):
    """GET /api/leaderboard response."""

    ranking_basis: str = "total_xp"
    entries: list[LeaderboardEntry]
    current_user: LeaderboardEntry
